taichi/program/program.cpp

```diff
@@ -73,53 +73,44 @@ Program::Program(Arch desired_arch)
 #ifdef TI_WITH_LLVM
     program_impl_ = std::make_unique<LlvmProgramImpl>(config, profiler.get());
 #else
-    TI_NOT_IMPLEMENTED
+    TI_ERROR("This taichi is not compiled with LLVM");
 #endif
   } else if (config.arch == Arch::metal) {
     if (!metal::is_metal_api_available()) {
-      TI_WARN("No Metal API detected.");
-      config.arch = host_arch();
+      TI_ERROR("No Metal API detected.");
```

Comment:
Shall we just do TI_ASSERT(metal::is_metal_api_available()) and get rid of the if-else? ;)